# Mathematical Statistics (Wackerly, 7th Edition)
## Chapter 2: Supplementary Exercises

**Textbook:** Mathematical Statistics with Applications, 7th Edition
**Authors:** Dennis Wackerly, William Mendenhall, Richard Scheaffer

---

### Exercises Covered
- 2.143: Complement rule for conditional probabilities
- 2.144: Sample points, events, and set operations
- 2.145: Counting sequences (permutations)
- 2.146: Probability all 5 cards same suit
- 2.147: Probability of a full house after redraw

Working through these using my **Universe → Intersection → Squeeze** framework and verifying results computationally.

In [2]:
import numpy as np
import matplotlib.pyplot as plt
from math import comb, factorial
from itertools import combinations

---

## Exercise 2.143: Complement Rule for Conditional Probabilities

### The Problem

**Show that Theorem 2.7 holds for conditional probabilities.** That is, if $P(B) > 0$, then:

$$P(\bar{A} | B) = 1 - P(A | B)$$

### My Approach

This is about proving that the complement rule I already know from unconditional probability still works when I "zoom into" the world of B.

**Step 1: Start from the definition**

By definition of conditional probability:

$$P(\bar{A} | B) = \frac{P(\bar{A} \cap B)}{P(B)}$$

**Step 2: Decompose B using a partition**

Here's the key insight — I can split B into two disjoint pieces:
- The part of B that's inside A: $A \cap B$
- The part of B that's outside A: $\bar{A} \cap B$

These are disjoint and together they make all of B:

$$B = (A \cap B) \cup (\bar{A} \cap B)$$

So by the addition rule for disjoint events:

$$P(B) = P(A \cap B) + P(\bar{A} \cap B)$$

**Step 3: Solve for the piece I need**

Rearranging:

$$P(\bar{A} \cap B) = P(B) - P(A \cap B)$$

**Step 4: Substitute back into the definition**

$$P(\bar{A} | B) = \frac{P(\bar{A} \cap B)}{P(B)} = \frac{P(B) - P(A \cap B)}{P(B)}$$

$$= \frac{P(B)}{P(B)} - \frac{P(A \cap B)}{P(B)} = 1 - P(A | B)$$

$$\boxed{P(\bar{A} | B) = 1 - P(A | B)}$$

### Why This Makes Sense

Once I've zoomed into B's world, the probabilities inside that world must still add up to 1. Event A either happens or it doesn't — even inside the world of B. So the conditional probabilities of A and $\bar{A}$ given B must be complements, just like unconditional probabilities.

**The B-world is a valid probability space on its own.** All the standard rules (including complements) still apply inside it.

In [3]:
# Numerical verification using the medical test example from my notes

# Setup (same as my Bayes' Theorem example)
P_D = 0.01       # P(Disease)
P_D_c = 0.99     # P(No Disease)
P_pos_D = 0.90   # P(+|Disease)
P_pos_D_c = 0.05 # P(+|No Disease)

# Compute P(+) using Law of Total Probability
P_pos = P_pos_D * P_D + P_pos_D_c * P_D_c  # = 0.0585

# P(D|+) using Bayes
P_D_given_pos = (P_pos_D * P_D) / P_pos

# Now verify the complement rule: P(D̅|+) should equal 1 - P(D|+)
P_D_bar_given_pos = (P_pos_D_c * P_D_c) / P_pos  # Direct calculation
complement_check = 1 - P_D_given_pos               # Using complement rule

print("Verification: Complement Rule for Conditional Probabilities")
print("=" * 58)
print(f"P(D|+)         = {P_D_given_pos:.6f}")
print(f"P(D̅|+) direct  = {P_D_bar_given_pos:.6f}")
print(f"1 - P(D|+)     = {complement_check:.6f}")
print()
print(f"Match? {np.isclose(P_D_bar_given_pos, complement_check)}")
print(f"Sum check: P(D|+) + P(D̅|+) = {P_D_given_pos + P_D_bar_given_pos:.6f} (should be 1.0)")

Verification: Complement Rule for Conditional Probabilities
P(D|+)         = 0.153846
P(D̅|+) direct  = 0.846154
1 - P(D|+)     = 0.846154

Match? True
Sum check: P(D|+) + P(D̅|+) = 1.000000 (should be 1.0)


---

## Exercise 2.144: Sample Points, Events, and Set Operations

### The Problem

Let $S$ contain four sample points: $E_1, E_2, E_3, E_4$.

**(a)** List all possible events in $S$ (include the null event).

**(b)** Use the identity $\sum_{i=0}^{n} \binom{n}{i} = 2^n$ to give the total number of events in $S$.

**(c)** Let $A = \{E_1, E_2, E_3\}$ and $B = \{E_2, E_4\}$. Give the sample points in: $A \cup B$, $A \cap B$, $\bar{A} \cap \bar{B}$, and $\bar{A} \cup \bar{B}$.

### Part (a): All Possible Events

An **event** is any subset of the sample space $S$. This includes:
- The **null event** $\emptyset$ (nothing happens)
- **Simple events** (single sample points)
- **Compound events** (multiple sample points)
- The **entire sample space** $S$ (something happens)

Let me list them all systematically by size:

In [4]:
# Part (a): Enumerate all possible events (subsets) of S

S = {'E1', 'E2', 'E3', 'E4'}

all_events = []
for size in range(len(S) + 1):
    for subset in combinations(sorted(S), size):
        all_events.append(set(subset))

print("All possible events in S = {E1, E2, E3, E4}:")
print("=" * 50)
for i, event in enumerate(all_events):
    label = "∅" if len(event) == 0 else str(sorted(event))
    size_label = f"(size {len(event)})"
    print(f"  {i+1:2d}. {label:40s} {size_label}")

print(f"\nTotal number of events: {len(all_events)}")

All possible events in S = {E1, E2, E3, E4}:
   1. ∅                                        (size 0)
   2. ['E1']                                   (size 1)
   3. ['E2']                                   (size 1)
   4. ['E3']                                   (size 1)
   5. ['E4']                                   (size 1)
   6. ['E1', 'E2']                             (size 2)
   7. ['E1', 'E3']                             (size 2)
   8. ['E1', 'E4']                             (size 2)
   9. ['E2', 'E3']                             (size 2)
  10. ['E2', 'E4']                             (size 2)
  11. ['E3', 'E4']                             (size 2)
  12. ['E1', 'E2', 'E3']                       (size 3)
  13. ['E1', 'E2', 'E4']                       (size 3)
  14. ['E1', 'E3', 'E4']                       (size 3)
  15. ['E2', 'E3', 'E4']                       (size 3)
  16. ['E1', 'E2', 'E3', 'E4']                 (size 4)

Total number of events: 16


### Part (b): Total Number of Events Using the Binomial Identity

The identity says:

$$\sum_{i=0}^{n} \binom{n}{i} = 2^n$$

**Why does this count events?** Each term $\binom{n}{i}$ counts the number of subsets of size $i$. So the sum counts **all subsets of all sizes** — which is exactly the number of possible events.

For $n = 4$:

$$\binom{4}{0} + \binom{4}{1} + \binom{4}{2} + \binom{4}{3} + \binom{4}{4} = 1 + 4 + 6 + 4 + 1 = 16 = 2^4$$

**Alternative intuition:** For each of the 4 sample points, I have 2 choices — include it in the event or don't. So total events = $2 \times 2 \times 2 \times 2 = 2^4 = 16$.

In [5]:
# Part (b): Verify using the binomial identity

n = 4

# Compute each term
terms = [comb(n, i) for i in range(n + 1)]
binomial_sum = sum(terms)

print("Binomial identity verification:")
print("=" * 45)
for i in range(n + 1):
    print(f"  C({n},{i}) = {terms[i]}")
print(f"\n  Sum = {' + '.join(str(t) for t in terms)} = {binomial_sum}")
print(f"  2^{n} = {2**n}")
print(f"\n  Match? {binomial_sum == 2**n}")

Binomial identity verification:
  C(4,0) = 1
  C(4,1) = 4
  C(4,2) = 6
  C(4,3) = 4
  C(4,4) = 1

  Sum = 1 + 4 + 6 + 4 + 1 = 16
  2^4 = 16

  Match? True


### Part (c): Set Operations

Given:
- $A = \{E_1, E_2, E_3\}$
- $B = \{E_2, E_4\}$
- $S = \{E_1, E_2, E_3, E_4\}$

**Computing each:**

**$A \cup B$** (everything in A OR B):
$$A \cup B = \{E_1, E_2, E_3\} \cup \{E_2, E_4\} = \{E_1, E_2, E_3, E_4\} = S$$

**$A \cap B$** (everything in BOTH A and B):
$$A \cap B = \{E_1, E_2, E_3\} \cap \{E_2, E_4\} = \{E_2\}$$

**$\bar{A} \cap \bar{B}$** (in neither A nor B):

First, find the complements:
- $\bar{A} = S \setminus A = \{E_4\}$
- $\bar{B} = S \setminus B = \{E_1, E_3\}$

$$\bar{A} \cap \bar{B} = \{E_4\} \cap \{E_1, E_3\} = \emptyset$$

**De Morgan's check:** $\bar{A} \cap \bar{B} = \overline{A \cup B} = \bar{S} = \emptyset$ ✓

**$\bar{A} \cup \bar{B}$** (outside A OR outside B):

$$\bar{A} \cup \bar{B} = \{E_4\} \cup \{E_1, E_3\} = \{E_1, E_3, E_4\}$$

**De Morgan's check:** $\bar{A} \cup \bar{B} = \overline{A \cap B} = \overline{\{E_2\}} = \{E_1, E_3, E_4\}$ ✓

In [ ]:
# Part (c): Verify set operations with Python

S = {'E1', 'E2', 'E3', 'E4'}
A = {'E1', 'E2', 'E3'}
B = {'E2', 'E4'}

A_bar = S - A
B_bar = S - B

results = {
    'A ∪ B':       A | B,
    'A ∩ B':       A & B,
    'Ā ∩ B̄':      A_bar & B_bar,
    'Ā ∪ B̄':      A_bar | B_bar,
}

print("Set Operations with A = {E1, E2, E3}, B = {E2, E4}")
print("=" * 55)
print(f"  Ā = S \\ A = {sorted(A_bar)}")
print(f"  B̄ = S \\ B = {sorted(B_bar)}")
print()
for name, result in results.items():
    display = "∅" if len(result) == 0 else str(sorted(result))
    print(f"  {name:8s} = {display}")

# Verify De Morgan's laws
print("\nDe Morgan's Laws Verification:")
print(f"  Ā ∩ B̄  == (A ∪ B)̄ ?  {A_bar & B_bar == S - (A | B)}")
print(f"  Ā ∪ B̄  == (A ∩ B)̄ ?  {A_bar | B_bar == S - (A & B)}")

In [ ]:
# Visualizing the set operations

import matplotlib.pyplot as plt
import matplotlib.patches as patches

fig, axes = plt.subplots(1, 4, figsize=(16, 4))

# Helper: draw S with colored sample points
def draw_venn(ax, title, highlighted, color='lightgreen'):
    ax.set_xlim(-0.5, 4.5)
    ax.set_ylim(-0.5, 2.5)
    ax.set_aspect('equal')
    ax.axis('off')
    ax.set_title(title, fontsize=11, fontweight='bold')
    
    # Draw S boundary
    rect = patches.FancyBboxPatch((0, 0), 4, 2, boxstyle="round,pad=0.1",
                                   linewidth=2, edgecolor='black', facecolor='white')
    ax.add_patch(rect)
    ax.text(3.8, 1.8, 'S', fontsize=10, fontweight='bold')
    
    # Sample points positions
    positions = {'E1': (0.8, 1.3), 'E2': (2, 1.3), 'E3': (3.2, 1.3), 'E4': (2, 0.5)}
    
    for name, (x, y) in positions.items():
        in_set = name in highlighted
        c = patches.Circle((x, y), 0.3, facecolor=color if in_set else 'lightgray',
                           edgecolor='black', linewidth=2)
        ax.add_patch(c)
        ax.text(x, y, name, ha='center', va='center', fontsize=9, fontweight='bold')

draw_venn(axes[0], 'A ∪ B = {E1,E2,E3,E4}', {'E1','E2','E3','E4'}, '#2ecc71')
draw_venn(axes[1], 'A ∩ B = {E2}', {'E2'}, '#3498db')
draw_venn(axes[2], 'Ā ∩ B̄ = ∅', set(), '#e74c3c')
draw_venn(axes[3], 'Ā ∪ B̄ = {E1,E3,E4}', {'E1','E3','E4'}, '#f39c12')

plt.tight_layout()
plt.show()

---

## Exercise 2.145: Counting Sequences for Physical Examination

### The Problem

A patient must have **18 checks or tests** performed. The sequence matters (time between tests varies depending on order). How many sequences would an efficiency expert need to study?

### My Approach

This is a **permutation** problem. I need to arrange 18 distinct tests in a sequence.

- First slot: 18 choices
- Second slot: 17 choices
- Third slot: 16 choices
- ...
- Last slot: 1 choice

Total sequences:

$$18! = 18 \times 17 \times 16 \times \cdots \times 2 \times 1$$

This is the **mn rule** applied repeatedly — at each step, the number of remaining choices decreases by one.

In [6]:
# Exercise 2.145: Compute 18!

n_tests = 18
total_sequences = factorial(n_tests)

print(f"Number of possible sequences for {n_tests} tests:")
print(f"  {n_tests}! = {total_sequences:,}")
print()
print(f"That's approximately {total_sequences:.3e}")
print()

# Put this in perspective
seconds_per_sequence = 1  # if we could evaluate 1 sequence per second
years = total_sequences / (60 * 60 * 24 * 365.25)
print("To put this in perspective:")
print(f"  If the expert could evaluate 1 sequence per second,")
print(f"  it would take approximately {years:.2e} years.")
print(f"  The universe is only ~1.38 × 10^10 years old.")

Number of possible sequences for 18 tests:
  18! = 6,402,373,705,728,000

That's approximately 6.402e+15

To put this in perspective:
  If the expert could evaluate 1 sequence per second,
  it would take approximately 2.03e+08 years.
  The universe is only ~1.38 × 10^10 years old.


---

## Exercise 2.146: Probability All 5 Cards Same Suit

### The Problem

Five cards are drawn from a standard 52-card deck. What is the probability that all 5 cards are of the **same suit**?

### My Approach: Universe → Favorable → Divide

**Step 1: Universe perspective**

Total ways to draw 5 cards from 52 (order doesn't matter):

$$|S| = \binom{52}{5}$$

**Step 2: Count favorable outcomes**

For all 5 cards to be the same suit:
- **Choose the suit:** 4 options (spades, hearts, diamonds, clubs) → $\binom{4}{1} = 4$
- **Choose 5 cards from that suit's 13:** $\binom{13}{5}$ ways

By the mn rule:

$$\text{Favorable} = \binom{4}{1} \times \binom{13}{5} = 4 \times 1287 = 5148$$

**Step 3: Compute the probability**

$$P(\text{all same suit}) = \frac{\binom{4}{1} \times \binom{13}{5}}{\binom{52}{5}} = \frac{5148}{2598960}$$

In [ ]:
# Exercise 2.146: Exact computation + Monte Carlo verification

# Exact calculation
favorable = comb(4, 1) * comb(13, 5)
total = comb(52, 5)
P_same_suit = favorable / total

print("Exercise 2.146: All 5 Cards Same Suit")
print("=" * 50)
print(f"  C(13,5) = {comb(13,5)}")
print(f"  Favorable = 4 × C(13,5) = 4 × {comb(13,5)} = {favorable}")
print(f"  Total     = C(52,5) = {total:,}")
print(f"  P(all same suit) = {favorable}/{total} ≈ {P_same_suit:.6f}")
print(f"                   ≈ {P_same_suit*100:.4f}%")
print(f"                   ≈ 1 in {1/P_same_suit:.0f}")

# Monte Carlo verification
rng = np.random.default_rng(42)
n_simulations = 500_000

# Build a deck: 4 suits × 13 cards each (suit encoded as card // 13)
deck = np.arange(52)
same_suit_count = 0

for _ in range(n_simulations):
    hand = rng.choice(deck, size=5, replace=False)
    suits = hand // 13
    if np.all(suits == suits[0]):
        same_suit_count += 1

P_monte_carlo = same_suit_count / n_simulations

print(f"\nMonte Carlo verification ({n_simulations:,} simulations):")
print(f"  Simulated P(all same suit) ≈ {P_monte_carlo:.6f}")
print(f"  Exact P(all same suit)     = {P_same_suit:.6f}")
print(f"  Difference                 = {abs(P_monte_carlo - P_same_suit):.6f}")

**Intuition check:** About 1 in 505 five-card hands has all cards from the same suit. That feels right — it's rare but not astronomically so. There are 4 suits to "aim for" but each suit only has 13 cards out of 52, so getting all 5 from one suit is quite restrictive.

---

## Exercise 2.147: Probability of a Full House After Redraw

### The Problem

A gambler is dealt five cards: **two aces, one king, one five, and one nine**. He discards the 5 and the 9 and is dealt **two more cards**. What is the probability of ending up with a **full house**?

### My Setup

**What the gambler keeps:** 2 aces + 1 king (3 cards)

**What he needs from the draw:** 2 more cards from the remaining deck

**Remaining deck:** $52 - 5 = 47$ cards (the original 5 dealt cards are all out of the deck)

**What's a full house?** Three of one kind + two of another. Starting from 2 aces and 1 king, there are exactly two ways to get a full house:

| Path | Draw | Result |
|------|------|--------|
| (a)  | 1 ace + 1 king | 3 aces + 2 kings ✓ |
| (b)  | 2 kings | 2 aces + 3 kings ✓ |

No other combination works because he'd need cards that match what he already holds.

### My Approach: Universe → Favorable → Divide

**Step 1: Universe perspective**

Total ways to draw 2 cards from the remaining 47:

$$|S| = \binom{47}{2}$$

**Step 2: Count favorable outcomes for each path**

**Path (a): Draw 1 ace and 1 king**

- Aces remaining in deck: $4 - 2 = 2$ aces
- Kings remaining in deck: $4 - 1 = 3$ kings
- Ways: $\binom{2}{1} \times \binom{3}{1} = 2 \times 3 = 6$

**Path (b): Draw 2 kings**

- Kings remaining in deck: 3
- Ways: $\binom{3}{2} = 3$

**Step 3: Total favorable (paths are disjoint, so I add)**

$$\text{Favorable} = 6 + 3 = 9$$

**Step 4: Compute the probability**

$$P(\text{full house}) = \frac{9}{\binom{47}{2}} = \frac{9}{1081}$$

In [ ]:
# Exercise 2.147: Exact computation + Monte Carlo verification

# Exact calculation
aces_remaining = 2   # 4 - 2 held
kings_remaining = 3  # 4 - 1 held
remaining_deck = 47  # 52 - 5 dealt

path_a = comb(2, 1) * comb(3, 1)  # 1 ace + 1 king
path_b = comb(3, 2)                # 2 kings
favorable = path_a + path_b
total = comb(remaining_deck, 2)
P_full_house = favorable / total

print("Exercise 2.147: Full House After Redraw")
print("=" * 50)
print(f"  Held: 2 aces, 1 king")
print(f"  Remaining deck: {remaining_deck} cards")
print(f"  Total ways to draw 2: C(47,2) = {total}")
print()
print(f"  Path (a) - 1 ace + 1 king: C(2,1)×C(3,1) = {path_a}")
print(f"  Path (b) - 2 kings:         C(3,2)         = {path_b}")
print(f"  Total favorable:                             {favorable}")
print()
print(f"  P(full house) = {favorable}/{total} ≈ {P_full_house:.6f}")
print(f"                ≈ {P_full_house*100:.4f}%")
print(f"                ≈ 1 in {1/P_full_house:.0f}")

# Monte Carlo verification
rng = np.random.default_rng(42)
n_simulations = 1_000_000

# Encode cards: each card has a rank (0-12) and suit (0-3)
# Held cards: 2 specific aces + 1 specific king
# We only care about ranks for full house detection

# Build the remaining deck of 47 cards (remove 2 aces, 1 king, 1 five, 1 nine)
full_deck = [(rank, suit) for rank in range(13) for suit in range(4)]
# Remove: 2 aces (rank 0), 1 king (rank 12), 1 five (rank 4), 1 nine (rank 8)
held_aces = [(0, 0), (0, 1)]       # Two aces (suits 0,1)
held_king = [(12, 0)]               # One king (suit 0)
discarded = [(4, 0), (8, 0)]        # Five and nine
removed = set(held_aces + held_king + discarded)
remaining = [card for card in full_deck if card not in removed]

# Held hand ranks
held_ranks = [0, 0, 12]  # 2 aces, 1 king

full_house_count = 0
for _ in range(n_simulations):
    draw_indices = rng.choice(len(remaining), size=2, replace=False)
    drawn_ranks = [remaining[i][0] for i in draw_indices]
    
    # Final hand ranks
    final_ranks = held_ranks + drawn_ranks
    
    # Check for full house: exactly one rank with count 3 and one with count 2
    from collections import Counter
    counts = sorted(Counter(final_ranks).values(), reverse=True)
    if counts == [3, 2]:
        full_house_count += 1

P_mc = full_house_count / n_simulations

print(f"\nMonte Carlo verification ({n_simulations:,} simulations):")
print(f"  Simulated P(full house) ≈ {P_mc:.6f}")
print(f"  Exact P(full house)     = {P_full_house:.6f}")
print(f"  Difference              = {abs(P_mc - P_full_house):.6f}")

**Intuition check:** About a 0.83% chance — roughly 1 in 120. That's actually not terrible for a poker redraw scenario. The gambler already has a strong foundation (pair of aces + a king), so he only needs specific cards from a relatively small pool.

**Key detail I want to remember:** The remaining deck is 47 cards, not 50. All 5 originally dealt cards are removed from the deck, including the discarded 5 and 9. Those cards are "spent" — they don't go back in.

---

## Summary of Results

| Exercise | Problem | Answer |
|----------|---------|--------|
| 2.143 | $P(\bar{A}\|B) = 1 - P(A\|B)$ | Proved using $B = (A \cap B) \cup (\bar{A} \cap B)$ decomposition |
| 2.144a | List all events in $S = \{E_1, E_2, E_3, E_4\}$ | 16 events (all subsets) |
| 2.144b | Total number of events | $2^4 = 16$ via $\sum \binom{4}{i} = 2^4$ |
| 2.144c | Set operations | $A \cup B = S$, $A \cap B = \{E_2\}$, $\bar{A} \cap \bar{B} = \emptyset$, $\bar{A} \cup \bar{B} = \{E_1, E_3, E_4\}$ |
| 2.145 | Sequences of 18 tests | $18! = 6{,}402{,}373{,}705{,}728{,}000$ |
| 2.146 | All 5 cards same suit | $\frac{4 \times \binom{13}{5}}{\binom{52}{5}} \approx 0.00198$ |
| 2.147 | Full house after redraw | $\frac{9}{\binom{47}{2}} = \frac{9}{1081} \approx 0.00832$ |